tests/library/test_json.py

```diff
@@ -597,6 +596,120 @@ def test_anyOf_objects(self, target_obj):
         # The actual check
         _generate_and_check(target_obj, schema_obj)
 
+class TestEnum:
+    simple_schema = """{
+        "enum": [1,"2",false]
+    }
+    """
+
+    @pytest.mark.parametrize("target_obj", [1, "2", False])
+    def test_enum(self, target_obj):
+        # First sanity check what we're setting up
+        schema_obj = json.loads(self.simple_schema)
+        validate(instance=target_obj, schema=schema_obj)
+
+        # The actual check
+        _generate_and_check(target_obj, schema_obj)
+
+    @pytest.mark.parametrize("bad_obj", ["1", 2, True])
+    def test_bad_enum(self, bad_obj):
+        schema_obj = json.loads(self.simple_schema)
+        grammar = gen_json(schema_obj)
+        bad_str = _to_compact_json(bad_obj)
+        with pytest.raises(ParserException):
```

Comment:
Does the exception embed the character on which it failed? If so, can we check that?

Reply:
I will add that check to enum -- nice idea!

As for capturing the `current_byte` (you are right -- I forgot the exception had one of those glued on!), what do you think of this for use in these tests? 

```python
def _check_match_failure(bad_string, failure_byte, schema_obj):
    grammar = gen_json(schema_obj)
    with pytest.raises(ParserException):
        try:
            grammar.match(bad_string, raise_exceptions=True)
        except ParserException as e:
            assert e.current_byte == failure_byte
            raise e
```

Should we unify all of the negative tests to use the Mock engine somehow, unify them to all use something like this ^, or do you think it's fine that we have different approaches  to negative tests here?

Reply:
That's roughly what I meant, but with `pytest.raises` you don't need `try...except`. It's something more like:
```python
with pytest.raises(ParserException) as pe:
   grammar.match(bad_string, raise_exceptions=True)
assert pe.value.current_byte == failure_byte
```
But that is the pattern which should be used for all the negative tests.

Reply:
Ah, I didn't realize the `pytest.raises` context manager actually gave you a handle on the exception, but of course it does. Thanks for the help -- I'll update in a bit

Reply:
Fixed.